lib/bumbler/hooks.rb

```diff
@@ -5,12 +5,6 @@ module Hooks
     @started_items = {}
     @slow_requires = {}
 
-    module RequireLogger
-      def require(path, *args)
```

Comment:
can we just set `private` here instead ?

Reply:
If you make the prepended function private, then you make `Kernel.require` private, which a lot of code expects to be public. module_function is what gives you the weird public module function/private behavior. You might be able to simulate it by having a separate mixin for the module prepend and the singleton class prepend...or something else that I'm not thinking of--I did not try that, but I can play around with it. But adding `private` was the first thing I tried and it broke a lot of gems that call `Kernel.require` directly. The other thing I noticed about prepend is that it doesn't play super nice with `watch_require!`. If something else prepends to Kernel, subsequent prepend calls in here just get ignored. Can try to figure something out, or open to suggestions if you get something that works.

Reply:
2 different modules sounds alright ... or use the same and call `private_class_method :require` after ?